lib/hdprivatekey.js

```diff
@@ -63,6 +63,58 @@ function HDPrivateKey(arg) {
 }
 
 /**
+ * Verifies that a given path is valid.
+ *
+ * @param {string|number} arg
+ * @param {boolean?} hardened
+ * @return {boolean}
+ */
+HDPrivateKey.isValidPath = function(arg, hardened) {
+  if (_.isString(arg)) {
+    var indexes = HDPrivateKey._getDerivationIndexes(arg);
+    return indexes !== null && _.all(indexes, HDPrivateKey.isValidPath);
+  }
+
+  if (_.isNumber(arg)) {
+    if (arg < HDPrivateKey.Hardened && hardened === true) {
+      arg += HDPrivateKey.Hardened;
+    }
+    return arg >= 0 && arg < HDPrivateKey.MaxIndex;
+  }
+
+  return false;
+};
+
+/**
+ * Internal function that splits a string path into a derivation index array.
+ * It will return null if the string path is malformed.
+ * It does not validates if a indexes are in bounds.
```

Comment:
So, this looks like it just splits the paths. The validation seems to be in the `isValidPath` function, it checks in the line 82 for `arg >= 0 && arg < HDPrivateKey.MaxIndex;` As it's a internal function it looks OK to me.